Assets/MixedRealityToolkit/MSBuild/Scripts/CSProjectInfo.cs

```diff
@@ -0,0 +1,363 @@
+        }
+
+        private void PopulateSupportedPlatformBuildConditions(List<string> supportedPlatformBuildConditions, string template, string configuration, IReadOnlyDictionary<BuildTarget, CompilationPlatformInfo> platforms)
+        {
+            foreach (KeyValuePair<BuildTarget, CompilationPlatformInfo> platform in platforms)
+            {
+                supportedPlatformBuildConditions.Add(Utilities.ReplaceTokens(template, new Dictionary<string, string>()
+                {
+                    { "##SUPPORTED_CONFIGURATION_TOKEN##", configuration },
+                    { "##SUPPORTED_PLATFORM_TOKEN##", platform.Value.Name }
+                }));
+            }
+        }
+
+        private string CreateProjectReferencesSet(string template, HashSet<string> additionalSearchPaths, bool inEditor)
+        {
+            if (Utilities.TryGetXMLTemplate(template, "PROJECT_REFERENCE", out string projectReferenceTemplate)
+                && Utilities.TryGetXMLTemplate(template, "PLUGIN_REFERENCE", out string pluginReferenceTemplate))
+            {
+                List<string> projectReferences = new List<string>();
+                foreach (CSProjectDependency<CSProjectInfo> dependency in csProjectDependencies)
+                {
+                    List<string> platformConditions = GetPlatformConditions(inEditor ? InEditorPlatforms : PlayerPlatforms, inEditor ? dependency.InEditorSupportedPlatforms : dependency.PlayerSupportedPlatforms);
+
+                    projectReferences.Add(Utilities.ReplaceTokens(projectReferenceTemplate, new Dictionary<string, string>()
+                    {
+                        { "##REFERENCE_TOKEN##", $"{dependency.Dependency.Name}.csproj" },
+                        { "<!--HINT_PATH_TOKEN-->", dependency.Dependency.ReferencePath.AbsolutePath },
+                        { "##CONDITION_TOKEN##", platformConditions.Count == 0 ? "false" : string.Join(" OR ", platformConditions)}
+                    }));
+                }
+
+                List<string> pluginReferences = new List<string>();
+                foreach (CSProjectDependency<PluginAssemblyInfo> dependency in pluginAssemblyDependencies)
+                {
+                    if (dependency.Dependency.Type == PluginType.Native)
+                    {
+                        //TODO Native plugins aren't yet supported
+                        continue;
+                    }
+                    List<string> platformConditions = GetPlatformConditions(inEditor ? InEditorPlatforms : PlayerPlatforms, inEditor ? dependency.InEditorSupportedPlatforms : dependency.PlayerSupportedPlatforms);
+
+                    pluginReferences.Add(Utilities.ReplaceTokens(pluginReferenceTemplate, new Dictionary<string, string>()
+                    {
+                        { "##REFERENCE_TOKEN##", dependency.Dependency.Name },
+                        { "##CONDITION_TOKEN##", platformConditions.Count == 0 ? "'false'" : string.Join(" OR ", platformConditions)},
```

Comment:
>"'false'" [](start = 81, length = 9)

Do you know why this condition token needs to be double-encapsulated by " and ', but the one on line 287 doesn't?

Reply:
It doesn't, fixed :)

---
In reply to: [301800758](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r301800758) [](ancestors = 301800758)